### PR TITLE
[WIP] Module/Week 1

### PR DESCRIPTION
**Pull request by adrianmcli**
### Practical

- [x] Getting Started with Circom (syntax + zkrepl)
- [x] Write a circuit to prove input to hash
- [ ] Write a circuit to prove private key to public key
- [x] Write a circuit to prove inclusion in Merkle tree
- [ ] Write a circuit to prove set membership

### Study

- [x] Symmetric vs Asymmetric Encryption (AES, RSA)
- [x] Hash Functions, Merkle Trees
- [ ] Digital Signatures (Schnorr)
- [x] DLP-based Public-Key Cryptography (DLP, DH, Elgamal)
- [x] Cryptographic Commitments
- [ ] Intro to Elliptic Curve Cryptography
- [ ] BLS Signatures (keep this simple and black-boxed)
- [ ] Pairing-Based Cryptography (keep this simple and black-boxed)

### CODE REVIEW

**Comment by adrianmcli** [comment]
@NOOMA-42 has stated he will fill in the Practical (Circom) section of this Module. I will review and make my edits afterwards.

**Comment by adrianmcli** [comment]
The answer for writing a circuit to check output of hash:

```circom
pragma circom 2.1.6;

include "circomlib/poseidon.circom";

template HashProof() {
    signal input preimage;
    signal input hash;
    signal output hashOutput;

    component hasher = Poseidon(1);
    hasher.inputs[0] <== preimage;

    hashOutput <== hasher.out;

    // Enforce that the hash matches the given hash
    hash === hashOutput;
}

component main = HashProof();

/* INPUT = {
    "preimage": "12345",
    "hash": "4267533774488295900887461483015112262021273608761099826938271132511348470966"
} */
```

Instead of showing the above, we should break it down so that we force the student to come up with the above. We can give it to them in parts.

**Comment by NOOMA-42** [comment]
I removed set membership section because a very simple set membership is Merkle tree inclusion. We just go over in the previous section.

File: 2024/MODULE_1.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+:::info
+**🤔 Consider the following:**
+1. Can you describe the structure of a Merkle tree?
+2. How are Merkle trees used within the blockchain context?
+3. Why are Merkle trees useful for efficient and secure verification of large data structures?
+:::
+
+### Digital Signatures (Schnorr)
+
+TODO - add some more detail about Schnorr signatures specifically
```
Comment (NOOMA-42) [inline 2024/MODULE_1.md:200]: Is this done?

File: 2024/MODULE_1.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+**Elliptic Curve Digital Signature Algorithm (ECDSA)** extends the traditional Digital Signature Algorithm (DSA) by incorporating elliptic curves as well, enabling stronger security with shorter key lengths. This makes the algorithm more efficient and suitable for constrained environments. ECDSA is widely used in various secure communication protocols, including SSL/TLS.
+
+The following StackExchange answer goes over the differences between some of the most popular digital signature protocols. It also leads into our next topic, which is another type of digital signature called the Edwards-curve Digital Signature Algorithm (EdDSA).
+
+- [What is the difference between the RSA, DSA, and ECDSA keys that ssh uses?](https://askubuntu.com/a/1000928/733503)
+
+### BLS Signatures
+
+TODO - keep this simple and black-boxed
```
Comment (NOOMA-42) [inline 2024/MODULE_1.md:299]: Is this done?

File: 2024/MODULE_1.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+BLS (Boneh-Lynn-Shacham) signatures are a type of cryptographic signature scheme that allows for efficient aggregation of individual signatures into a single signature. It also makes use of pairing-based cryptography.
+
+The intuition of BLS is that it is an aggregation signature. The image below shows how the public key and signature are aggregated. The advantage is that all signatures can be verified at once. That's why it is commonly used in consensus protocols.
+![BLS](./assets/bls-signatures.jpeg)
+
+### Pairing-Based Cryptography
+
+TODO - keep this simple and black-boxed
```
Comment (NOOMA-42) [inline 2024/MODULE_1.md:308]: Is this done?

**Review by NOOMA-42 — CHANGES_REQUESTED** [review]

File: 2024/MODULE_1.md (hunk 2 above)
Comment (adrianmcli) [reply 2024/MODULE_1.md:299]: Not yet! If you'd like, you can move the content over and I can review.

File: 2024/MODULE_1.md (hunk 3 above)
Comment (adrianmcli) [reply 2024/MODULE_1.md:308]: @NOOMA-42 Do you feel like this is simple and black-boxed enough?

File: 2024/MODULE_1.md (hunk 2 above)
Comment (adrianmcli) [reply 2024/MODULE_1.md:299]: @NOOMA-42 Do you feel like this is simple and black-boxed enough?

**Comment by adrianmcli** [comment]
@NOOMA-42 do you think it is sufficiently simple and black-boxed?

I will just merge it now, we can always improve this later.

**Pull request merged**